sklearn/semi_supervised/label_propagation.py

```diff
@@ -264,12 +259,15 @@ def fit(self, X, y):
 
         l_previous = np.zeros((self.X_.shape[0], n_classes))
 
-        remaining_iter = self.max_iter
         unlabeled = unlabeled[:, np.newaxis]
         if sparse.isspmatrix(graph_matrix):
             graph_matrix = graph_matrix.tocsr()
-        while (_not_converged(self.label_distributions_, l_previous, self.tol)
-               and remaining_iter > 1):
+
+        self.n_iter_ = 0
+        while self.n_iter_ < self.max_iter:
```

Comment:
With the `for` loop, there are two alternatives:

## Alternative 1

Setting `self.n_iter_` after the `for` loop. It would look a bit ugly because we'll need to figure out whether in the last iteration the loop the tolerance condition was met (then `self.n_iter_ = n_iter`) or the `else:` branch was reached (in which case `self.n_iter_ = n_iter + 1`):

```python
self.n_iter_ = 0
for n_iter in range(self.max_iter):
    if converged:
        break
    # ...
else:
    # warn
    self.n_iter_ = 1 # Count the last iteration.

self.n_iter_ += n_iter
```

## Alternate 2

Leaving `self.n_iter_ += 1` inside the for-loop. In this case case the loop variable (i.e. `n_iter`) is not used anywhere.

```python
self.n_iter_ = 0
for n_iter in range(self.max_iter):
    if converged:
        break
    # ...
    self.n_iter_ += 1
else:
    # warn
```

----

Were any of these versions what you (both of you) had in mind? Personally, I like the `while` loop and then Alternative 2. :)